lib/assets/javascripts/turbograft/turbohead.coffee

```diff
@@ -76,37 +99,37 @@ noDatasetMatchesIn = (attribute, collection) ->
 updateLinkTags = (activeDocument, newLinks) ->
   # style tag load events don't work in all browsers
   # as such we just hope they load ¯\_(ツ)_/¯
-  newLinks.forEach((linkNode) -> insertLinkTask(activeDocument, linkNode)())
-
-updateScriptTags = (activeDocument, newScripts, callback) ->
-  asyncSeries(
-    newScripts.map((scriptNode) -> insertScriptTask(activeDocument, scriptNode)),
-    callback
+  newLinks.forEach((linkNode) ->
+    newNode = linkNode.cloneNode()
+    activeDocument.head.appendChild(newNode)
+    triggerEvent("page:after-link-inserted", newNode)
   )
 
-asyncSeries = (tasks, callback) ->
-  return callback() if tasks.length == 0
-  task = tasks.shift()
-  task(-> asyncSeries(tasks, callback))
+updateScriptTags = (activeDocument, newScripts) ->
+  promise = Promise.resolve()
+  newScripts.forEach (scriptNode) ->
```

Comment:
This can be made much cleaner with `Promise.all`, since it will be the same as `Promise.resolve` when called on an empty array.

```
scriptInsertions = newScripts.map(-> insertScript(activeDocument, scriptNode))
Promise.all(scriptInsertions)
```
or even
```
Promise.all(newScripts.map(-> insertScript(activeDocument, scriptNode)))
```
if you're feeling one linery.